XCUITests/DownloadFilesTests.swift

```diff
@@ -0,0 +1,174 @@
+        waitforExistence(app.tables["Context Menu"])
+        XCTAssertTrue(app.tables["Context Menu"].staticTexts[testFileName].exists)
+        XCTAssertTrue(app.tables["Context Menu"].cells["download"].exists)
+        app.buttons["Cancel"].tap()
+    }
+
+    func testDownloadOptionFromBrowserTabMenu() {
+        navigator.goto(BrowserTabMenu)
+        navigator.goto(HomePanel_Downloads)
+        XCTAssertTrue(app.tables["DownloadsTable"].exists)
+    }
+
+    func testDownloadFile() {
+        downloadFile(fileName: testFileName)
+        navigator.goto(BrowserTabMenu)
+        navigator.goto(HomePanel_Downloads)
+
+        waitforExistence(app.tables["DownloadsTable"])
+        // There should be one item downloaded. It's name and size should be shown
+        checkTheNumberOfDownloadedItems(items: 1)
+        XCTAssertTrue(app.tables.cells.staticTexts[testFileName].exists)
+        XCTAssertTrue(app.tables.cells.staticTexts[testFileSize].exists)
+    }
+
+    func testDeleteDownloadedFile() {
+        downloadFile(fileName: testFileName)
+        navigator.goto(BrowserTabMenu)
+        navigator.goto(HomePanel_Downloads)
+        waitforExistence(app.tables["DownloadsTable"])
+
+        deleteItem(itemName: testFileName)
+        waitforNoExistence(app.tables.cells.staticTexts[testFileName])
+
+        // After removing the number of items should be 0
+        checkTheNumberOfDownloadedItems(items: 0)
+    }
+
+    func testShareDownloadedFile() {
+        downloadFile(fileName: testFileName)
+        navigator.goto(BrowserTabMenu)
+        navigator.goto(HomePanel_Downloads)
+        app.tables.cells.staticTexts[testFileName].swipeLeft()
+
+        XCTAssertTrue(app.tables.cells.buttons["Share"].exists)
+        XCTAssertTrue(app.tables.cells.buttons["Delete"].exists)
+
+        app.tables.cells.buttons["Share"].tap()
+        waitforExistence(app.otherElements["ActivityListView"])
+        if iPad() {
+            app.otherElements["PopoverDismissRegion"].tap()
+        } else {
+            app.buttons["Cancel"].tap()
+        }
+    }
+
+    func testLongPressOnDownloadedFile() {
+        downloadFile(fileName: testFileName)
+        navigator.goto(BrowserTabMenu)
+        navigator.goto(HomePanel_Downloads)
+
+        waitforExistence(app.tables["DownloadsTable"])
+        app.tables.cells.staticTexts[testFileName].press(forDuration: 2)
+        waitforExistence(app.otherElements["ActivityListView"])
+        if iPad() {
+            app.otherElements["PopoverDismissRegion"].tap()
+        } else {
+            app.buttons["Cancel"].tap()
+        }
+     }
+
+    private func downloadFile(fileName: String) {
+        navigator.openURL(testURL)
+        waitUntilPageLoad()
+
+        app.webViews.staticTexts[fileName].tap()
+        app.webViews.buttons["Download"].tap()
+        waitforExistence(app.tables["Context Menu"])
+        app.tables["Context Menu"].cells["download"].tap()
+    }
+
+    func testDownloadMoreThanOneFile() {
```

Comment:
perhaps it could be used in the future if this takes a parameter and download x number of files?

Reply:
So you mean to modify the `private func downloadFile(fileName: testFileName)` function to something like:
`private func downloadFileXtimes(fileName: String, numberOfDownloads: Int)`?

Reply:
yes, that would be more useful in the long run.

Reply:
👍